tests/Indexation/AbstractTypeTest.php

```diff
@@ -50,10 +55,70 @@ public function testItAcceptTypes()
         $fooType->update(new \stdClass(), 'foo');
     }
 
+    public function testItFlushes()
+    {
+        $type = $this->prophesize(Type::class);
+        $index = $this->prophesize(Index::class);
+        $responseSet = $this->prophesize(ResponseSet::class);
+
+        $fooType = new FooType($this->logger->reveal());
+        $fooType->setType($type->reveal());
+        $fooType->stageForInsert(['object1'], 1);
+        $fooType->stageForInsert(['object2'], 2);
+
+        $responseSet->hasError()->willReturn(false);
+        $type->addDocuments([
+            new Document(1, ['object1']),
+            new Document(2, ['object2'])
+        ])->shouldBeCalled()->willReturn($responseSet);
+
+        $type->getIndex()->willReturn($index);
+        $index->refresh()->shouldBeCalled();
+
+        $fooType->flush();
```

Comment:
An interesting test that is not present here is the fact that a second flush call will not trigger a call to `addDocuments`.

Reply:
Good catcth. I add the test

Reply:
test added